### ISSUE TITLE
syntax: handling of empty strings or trailing white spaces

### ISSUE THREAD

**Issue by langpavel** [projectfluent/fluent#223]
## Take this as an example:
```
form-name =
    {$user_gender ->
        [male] {"Ten "}
        [female] {"She name is "}
       *[other] {""}
    }{ $user_name }
```

## 1. Will be good if can be written as:
```
form-name =
    {$user_gender ->
        [male] His name is \
        [female] Her name is \
       *[other] \
    }{ $user_name }
```

## 2. Will be better if can be written as:
```
form-name =
    {$user_gender ->
        [male] His name is
        [female] Her name is
       *[other] \-
    } { $user_name }
```
The latest `\-` in `*[other]` is *trim*
Like this:
 - `\` ***EOL*** → emty string or keep spaces before
 - `\` `-` ***EOL*** → greedy trim (like eat all whitespaces around)
 - `\` `+` ***EOL*** → greedy+one space trim (like eat all whitespaces around, keep one if in middle)

**Comment by aphillips** [comment]
Hmm... these examples (and this syntax) promotes a common I18N mistake: the idea of sentence concatenation working equivalently across languages. This example assumes that the user name always occurs after the rest of the sentence. Promoting this kind of thinking _even if we know better than to use it ourselves_ is a bad thing. Ideally the syntax should discourage it as a behavior at all. After all `His name is {username}` in Japanese might be something like `彼の名前は{username}です` (I used google translate here).

Note too that @langpavel's choice about spaces has implications for translations of the file. A Japanese translator does not want that (hard to see!) space if it is meaningful. In other words, do this:

```
form-name = 
    {$user_gender ->
        [male] His name is {$user_name}
        [female] Her name is {$user_name}
       *[other] {$username}
    }
```

... and disallow concatenations. It answers the question at the same time :-)

**Comment by langpavel** [comment]
> This example assumes that the user name always occurs after the rest of the sentence.

No, this have nothing to do with placement

> in Japanese might be something like 彼の名前は{username}です

You can always write it as @aphillips proposed.

> In other words, do this:
```
form-name = 
    {$user_gender ->
        [male] His name is {$user_name}
        [female] Her name is {$user_name}
       *[other] {$username}
    }
```

Correct, but much of redundancy, I like this variant, but this may be not best solution always I guess... will think about.

**Comment by langpavel** [comment]
@aphillips good point about *the idea of sentence concatenation working equivalently across languages*

I do not expect, that `\-` and `\+` operators may cross scope of `Message`
This is not intended and I'm not sure if even can be implemented…

**Comment by eemeli** [comment]
Closing as WONTFIX; variants should really be written out as complete as possible, even if it ends up duplicating parts of the message multiple times. This will benefit translators, and won't actually increase file size almost at all, once you take compression into account.
